File: src/17장/17-2.py
n=int(input())
a=list(map(int,input().split()))
a.sort()

def binarySeach(target):
  start=0
  end=n-1
  while start<=end:
    mid=(start+end)//2
    if a[mid]==target:
      print(1)
      return
    elif a[mid]<=target:
      start=mid+1
    else:
      end=mid-1
  print(0)
  return



m=int(input())
b=list(map(int,input().split()))
for i in range(m):
  binarySeach(b[i])

File: src/17장/17-2.java
import java.util.*;
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
		Scanner sc = new Scanner(System.in);
		int N = sc.nextInt();
		int[] a = new int[N];
		for (int i = 0; i < N; i++) {
			a[i] = sc.nextInt();
		}
		Arrays.sort(a);
		int M = sc.nextInt();
		StringBuilder sb = new StringBuilder();
		for (int i = 0; i < M; i++) {
			if (binarySearch(a, sc.nextInt()) >= 0) {
				sb.append(1).append('\n');
			} else {
				sb.append(0).append('\n');
			}
		}
		System.out.println(sb);
	}

	public static int binarySearch(int[] arr, int target) {
		int lo = 0;
		int hi = arr.length - 1;
		while (lo <= hi) {
			int mid = (lo + hi) / 2;
			if (target < arr[mid])
				hi = mid - 1;
			else if (target > arr[mid])
				lo = mid + 1;
			else
				return mid;
		}
		return -1;
	}
}